nevergrad/optimization/optimizerlib.py

```diff
@@ -860,6 +860,44 @@ def _internal_provide_recommendation(self) -> tp.ArrayLike:
         return self.avg
 
 
+class RescaleOptimizer(base.Optimizer):
+    """Proposes a version of a base optimizer which works at a different scale."""
+    def __init__(
+            self,
+            parametrization: IntOrParameter,
+            budget: tp.Optional[int] = None,
+            num_workers: int = 1,
+            base_optimizer: base.OptCls = CMA,
+            scale: tp.Optional[float] = None,
+    ) -> None:
+        super().__init__(parametrization, budget=budget, num_workers=num_workers)
+        self.base_optimizer = base_optimizer(parametrization, budget=budget, num_workers=num_workers)
+        if scale is None:
+            assert budget is not None, "Either scale or budget must be known in RescaleOptimizer."
+            scale = np.sqrt(np.log(self.budget) / self.dimension)
+        self.scale = scale
+        assert self.scale != 0., "scale should be non-zero in Rescaler."
+
+    def rescale_candidate(self, candidate: p.Parameter, inverse: bool=False) -> p.Parameter:
+        data = candidate.get_standardized_data(reference=self.parametrization)
+        return (
+            self.parametrization.spawn_child().set_standardized_data(self.scale * data if not inverse
+            else (1. / self.scale) * data))
+
+    def _internal_ask_candidate(self) -> p.Parameter:
+        candidate = self.base_optimizer.ask()
+        candidate = self.rescale_candidate(candidate)
+        return candidate
+
+    def _internal_tell_candidate(self, candidate: p.Parameter, loss: tp.FloatLoss) -> None:
+        candidate = self.rescale_candidate(candidate)
```

Comment:
here is the problem: you can indeed create a new candidate to send to the user, but when you get back this candidate, you should not create yet another new candidate to send back to the algorithm, because that would be a tell-not-asked
You need to identify the initial candidate, you can do this by keeping a {sent_candidate.uid: initial_candidate}

Reply:
Done, thx.

Reply:
Hum, still stochastic, and I have no idea why.

Reply:
Will finish the rest first, that one can wait..

Reply:
is it stochastic or deterministic with a modified outcome?

Reply:
Now fixed.